Shell/LineEditor.cpp

```diff
@@ -37,6 +37,59 @@ void LineEditor::append(const String& string)
     m_cursor = m_buffer.size();
 }
 
+void LineEditor::tab_complete_first_token()
+{
+    auto input = String::copy(m_buffer);
+
+    String path = getenv("PATH");
+    if (path.is_empty())
+        return;
+    auto directories = path.split(':');
+
+    String match;
+
+    // Go through the files in PATH.
+    for (const auto& directory : directories) {
+        CDirIterator programs(directory.characters(), CDirIterator::SkipDots);
+        while (programs.has_next()) {
+            String program = programs.next_path();
+            if (program.starts_with(input)) {
```

Comment:
Okay, that's better. Let's also write this in early-return style to reduce nesting:

```cpp
if (!program.starts_with(input))
    continue;
```